common/planners/src/plan_builder.rs

```diff
@@ -106,6 +130,29 @@ impl PlanBuilder {
         let input_schema = self.plan.schema();
         let aggr_projection_fields = self.exprs_to_fields(&aggr_expr, &input_schema)?;
 
+        // Aggregator check.
+        for e_aggr in &aggr_expr {
+            // do not check literal expressions
+            if let ExpressionPlan::Literal(_) = e_aggr {
+                continue;
+            } else if !e_aggr.has_aggregator()? {
+                let mut in_group_by = false;
+                // Check if e_aggr is in group-by's list
+                for e_group in &group_expr {
+                    if aggr_group_expr_eq(&e_aggr, &e_group, &input_schema)? {
+                        in_group_by = true;
+                        break;
+                    }
+                }
+                if !in_group_by {
+                    return Result::Err(ErrorCodes::UnknownException(format!(
```

Comment:
We need a new error code type

Reply:
@zhang2014 
Hmm, this pr is a demo for @jyizheng , not a work pr.
Please review on:
https://github.com/datafuselabs/datafuse/pull/405
Thanks.